crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1190,6 +1190,13 @@ impl AuthorityStore {
                 &self.perpetual_tables.executed_effects,
                 iter::once(tx_digest),
             )?;
+        if let Some(events_digest) = effects.events_digest() {
+            write_batch = write_batch.delete_range(
+                &self.perpetual_tables.events,
+                &(*events_digest, usize::MIN),
+                &(*events_digest, usize::MAX),
```

Comment:
Strictly speaking we want &(*event_digest+1, usize::MIN) here, but I guess we won't have usize::MAX number of events in transaction...